python/pyspark/ml/clustering.py

```diff
@@ -291,6 +292,317 @@ def _create_model(self, java_model):
+            checkpoint (-1). E.g. 10 means that the cache will get checkpointed every 10 iterations.
+        :param maxIter:             Number of iterations. Default to 20
+        :param seed:                Param for random seed
+        __init__(self, featuresCol="features", k=10, \
+                 optimizer="online", learningOffset=1024.0, learningDecay=0.51, \
+                 subsamplingRate=0.05, optimizeDocConcentration=True, \
+                 checkpointInterval=10, maxIter=20, seed=None):
+        """
+        super(LDA, self).__init__()
+        self._java_obj = self._new_java_obj("org.apache.spark.ml.clustering.LDA", self.uid)
+        self.k = Param(self, "k", "number of clusters to create")
+        self.optimizer = Param(self, "optimizer",
+                               "LDA optimizer, Currently 'em', 'online' are supported.")
+        self.learningOffset = Param(self, "learningOffset",
+                                    "learning parameter that downweights early iterations")
+        self.learningDecay = Param(self, "learningDecay", "Learning rate")
+        self.subsamplingRate = Param(self, "subsamplingRate",
+                                     "Fraction of the corpus to be sampled")
+        self.optimizeDocConcentration = Param(self, "optimizeDocConcentration",
+                                              "Indicates whether the docConcentration \
+                                              will be optimized during training")
+        self._setDefault(k=10, optimizer="online", learningOffset=1024.0, learningDecay=0.51,
+                         subsamplingRate=0.05, optimizeDocConcentration=True,
+                         checkpointInterval=10, maxIter=20)
+        kwargs = self.__init__._input_kwargs
+        self.setParams(**kwargs)
+
+    def _create_model(self, java_model):
+        if self.getOptimizer() == "em":
+            return DistributedLDAModel(java_model)
+        else:
+            return LocalLDAModel(java_model)
+
+    @keyword_only
+    @since("2.0.0")
+    def setParams(self, featuresCol="features", k=10,
+                  optimizer="online", learningOffset=1024.0, learningDecay=0.51,
+                  subsamplingRate=0.05, optimizeDocConcentration=True,
+                  checkpointInterval=10, maxIter=20, seed=None):
+        """
+        setParams(self, featuresCol="features", k=10, \
+                  optimizer="online", learningOffset=1024.0, learningDecay=0.51, \
+                  subsamplingRate=0.05, optimizeDocConcentration=True, \
+                  checkpointInterval=10, maxIter=20, seed=None):
+
+        Sets params for LDA.
+        """
+        kwargs = self.setParams._input_kwargs
+        return self._set(**kwargs)
+
+    @since("2.0.0")
+    def setK(self, value):
+        """
+        Sets the value of :py:attr:`k`.
+
+        >>> algo = LDA().setK(10)
+        >>> algo.getK()
+        10
```

Comment:
I take a look at #10975, but think these code are not only test code, but can also be treated as sample code for user to refer. Sometimes code is more intuitive than comment. What do you think ?